src/js/stores/MesosSummaryStore.js

```diff
@@ -18,15 +17,31 @@ import StateSummary from '../structs/StateSummary';
 var TimeScales = require('../constants/TimeScales');
 import VisibilityStore from './VisibilityStore';
 
-var requestInterval = null;
+let requestInterval = null;
+let isInactive = false;
 
 function startPolling() {
   if (requestInterval == null && MesosSummaryStore.shouldPoll()) {
     // Should always retrieve bulk summary when polling starts
     MesosSummaryActions.fetchSummary(TimeScales.MINUTE);
-    requestInterval = setInterval(
-      MesosSummaryActions.fetchSummary, Config.getRefreshRate()
-    );
+
+    requestInterval = setInterval(function () {
+      let wasInactive = isInactive && !VisibilityStore.get('isInactive');
+      isInactive = VisibilityStore.get('isInactive');
+
+      if (!isInactive) {
+        if (wasInactive) {
+          // Flush history with new data set
+          MesosSummaryActions.fetchSummary(TimeScales.MINUTE);
+        } else {
+          MesosSummaryActions.fetchSummary();
+        }
+      } else {
+        // If not active, push null placeholder. This will ensure we maintain
+        // history when navigating back, for case where history server is down
+        MesosSummaryStore.processSummaryError({silent: true});
```

Comment:
Because we only want to push a summary on the stack without side effects (like re-rendering etc). The tab is out of focus so we don't want it to do any work. It only matters that there is appropriate history when we return focus to the tab.